R/lint.R

```diff
@@ -128,7 +128,8 @@ lint <- function(filename, linters = NULL, ..., cache = FALSE, parse_settings =
 lint_dir <- function(path = ".", ...,
                      relative_path = TRUE,
                      exclusions = list("renv", "packrat"),
-                     pattern = rex(".", one_of("Rr"), or("", "html", "md", "nw", "rst", "tex", "txt"), end),
+                     pattern = rex(".", or(group(one_of("Rr"), or("", "html", "md", "nw", "rst", "tex", "txt")),
```

Comment:
@AshesITR WDYT about making this case-insensitive ourselves by prepending `(?i)` to the `rex()` output?

AFAICT there's not a way to do that within {rex} itself: https://github.com/r-lib/rex/issues/85

That would make this a lot simpler: `rex(".", or("r", "rmd", "qmd", "rnw", "rhtml", "rrst", "rtex", "rtxt"), end)` then later `dir(pattern = paste0("(?i)", pattern))`.

Alternatively we can add a helper for this to hide the ugly/distracting details from the user: `pattern = r_files_pattern()`.

Reply:
This seems smart. I was wondering about a helper when I found myself writing that quite unwieldy line. Would it have been smart to raise that question in the issue before creating this RP? I'm still trying to get a handle on GH etiquette/best practice

Reply:
Thanks for asking!

I'm not sure there's a universal rule. Certainly it can't hurt to ask "Hey, it seems a helper would be more useful here, WDYT?" esp. along with a proposed implementation. Whether to go ahead and do so without suggestion is a bit more dicey.

Ultimately we as reviewers/maintainers will have the best understanding on if it's the right choice for the codebase -- if you're new to the codebase, it's possible there's something missing from the bigger picture, or there's just a style preference why things are done a certain way.

Reply:
@AshesITR gentle ping on the question above. I lean towards a helper (and exporting it), it will also be useful for users customizing lint patterns after #2135.

Reply:
Silently adding `(?i)` to used-supplied egexes seems like asking for trouble. For our default regex I think the only important thing is that the list of file extensions is easily visible to the user.

What about creating a helper `rex_file_exts("R", ..., ignore_case = TRUE)`?

Reply:
Note that filesystems are case-sensitive on all platforms except windows.

Reply:
I'm also fine with ditching rex() here until the issue is resolved and instead using a string regex.

Reply:
My memory is MacOS is also case-insensitive, which a quick google suggests is true (at least by default): https://discussions.apple.com/thread/251191099... anyway you're right. What I had in mind is silently applying that to the default argument, not user-supplied arguments, which we could do with `missing()` but better to avoid that.

`rex_file_exts()` looks reasonable enough.

Let's merge this as-is for now & handle as a follow-up? If so I'll write the NEWS.

Reply:
> I'm also fine with ditching rex() here until the issue is resolved and instead using a string regex.

one concern there is `(?i)` is not particularly common (?), so it might be (?) confusing to expose that in the default value.

Reply:
> > I'm also fine with ditching rex() here until the issue is resolved and instead using a string regex.
> 
> 
> 
> one concern there is `(?i)` is not particularly common (?), so it might be (?) confusing to expose that in the default value.

If that were the case, we'd to a (humble) job towards educating our users about it 😊
I expect R programmers to be able to google for `regex (?i)` if they don't understand and want to know.

Reply:
I was worried `(?i)` wouldn't google well but I am wrong :)